src/coreclr/jit/fgbasic.cpp

```diff
@@ -516,6 +562,155 @@ void Compiler::fgReplaceSwitchJumpTarget(BasicBlock* blockSwitch, BasicBlock* ne
+    while (i < succCount)
+    {
+        if (succTab[i] == oldSucc) // We will update when succTab[i] matches
+        {
+            // Remove the old edge [block => oldSucc]
+            //
+            fgRemoveAllRefPreds(oldSucc, block);
+
+            // Change the succTab entry to branch to the new location
+            //
+            succTab[i] = newSucc;
+
+            // Create the new edge [block => newSucc]
+            //
+            FlowEdge* const newEdge = fgAddRefPred(newSucc, block);
+
+            // Now set the correct value of newEdge's DupCount
+            // and replace any other successors in succTab[] that go to oldSucc.
+            //
+            i++;
+            while (i < succCount)
+            {
+                if (succTab[i] == oldSucc)
+                {
+                    // We also must update this entry in the succTab
+                    //
+                    succTab[i] = newSucc;
+                    newSucc->bbRefs++;
+                    newEdge->incrementDupCount();
+                }
+                i++; // Check the next entry in succTab[]
+            }
+            return;
+        }
+        i++; // Check the next entry in succTab[] for a match
+    }
+    noway_assert(!"Did not find oldTarget in succTab[]");
+}
+
+//------------------------------------------------------------------------
+// fgRemoveEhfSuccessor: update BBJ_EHFINALLYRET block to remove `succ` as a successor.
+// Updates the predecessor list of `succ`.
+//
+// Arguments:
+//   block   - BBJ_EHFINALLYRET block
+//   succ    - successor
+//
+void Compiler::fgRemoveEhfSuccessor(BasicBlock* block, BasicBlock* succ)
+{
+    assert(block != nullptr);
+    assert(succ != nullptr);
+    assert(fgPredsComputed);
+    assert(block->KindIs(BBJ_EHFINALLYRET));
+
+    // Don't `assert(succ->isBBCallAlwaysPairTail())`; we've already unlinked the CALLFINALLY
+    assert(succ->KindIs(BBJ_ALWAYS));
+
+    fgRemoveRefPred(succ, block);
+
+    BBehfDesc* const ehfDesc   = block->GetJumpEhf();
+    unsigned         succCount = ehfDesc->bbeCount;
+    BasicBlock**     succTab   = ehfDesc->bbeSuccs;
+    bool             found     = false;
+
+    // Walk the successor table looking for the specified successor block.
+    for (unsigned i = 0; i < succCount; i++)
+    {
+        if (succTab[i] == succ)
+        {
+            // If it's not the last one, move everything after in the table down one slot.
+            if (i + 1 < succCount)
+            {
+                memmove_s(&succTab[i], (succCount - i) * sizeof(BasicBlock*), &succTab[i + 1],
+                          (succCount - i - 1) * sizeof(BasicBlock*));
+            }
+
+            --succCount;
+
+            found = true;
+#ifdef DEBUG
+            // We only expect to see a successor once in the table.
+            for (; i < succCount; i++)
+            {
+                assert(succTab[i] != succ);
+            }
+#endif // DEBUG
```

Comment:
Break out of the loop here if not debug?

Reply:
I added the `break` but I don't like having the loop iterations of the main loop being debug dependent.